### PR TITLE
Cache improvements

### PR DESCRIPTION
**Pull request by 5K1PY**
Add cache versioning and some internal cleanup

### CODE REVIEW

**Review by SvizelPritula — CHANGES_REQUESTED** [review]
Wouldn't it be much easier to parse the yaml file with a library?

**Comment by 5K1PY** [comment]
> Wouldn't it be much easier to parse the yaml file with a library?

I don't really want to as this can raise all sort of errors.

File: pisek/jobs/cache.py [hunk 1]
```diff
@@ -27,15 +27,30 @@
...
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(version={self.version})"
+
+    @staticmethod
+    def read(f: TextIO) -> "CacheInfo":
+        tag = f.readline().removeprefix("- ").strip()
+        if tag != CacheInfo.yaml_tag:
+            return CacheInfo("?.?.?")
+        version = f.readline().removeprefix("  version:").strip()
+        return CacheInfo(version)
```
Comment (SvizelPritula) [inline pisek/jobs/cache.py:47]: Is it guaranteed that `CacheInfo` will be the first entry?

Comment (5K1PY) [reply pisek/jobs/cache.py:47]: For all new versions, yes.

**Review by SvizelPritula — APPROVED** [review]
LGTM

**Pull request merged**